In [6]:
import math
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import scipy.stats as stats
import statsmodels.formula.api as smf
import statsmodels.api as sm

from math import exp
from scipy.stats import t
from scipy.stats import f_oneway
from scipy.stats import chi2_contingency
from scipy.stats import ttest_rel
from scipy.stats import binom
from scipy.stats import poisson
from scipy.stats import ttest_1samp
from scipy.stats import mode
from scipy.stats import ttest_ind
from scipy.stats import norm


## Exercise I
Two students are counting the number of cars passing by on different stretches of road. They
assume that the number of cars passing by in specific time intervals follow Poisson distributions.
On the first road (road 1) they assume that the expected number of cars passing by is λ1 =
10/hour, while on the second road (road 2) they assume that the expected number of cars
passing by is λ2 = 15/hour.

Now they define two random variables:

• X1: number of cars passing by on road 1 in 15 minutes


• X2: number of cars passing by on road 2 in 10 minutes.

You can assume that X1 and X2 are independent.


#### Question I.1 (1)
What is the probability P(X1 = 10)?

In [7]:
λ1_15min = 10 * (15 / 60)  # rate for 15 minutes
k1 = 10  # number of cars we want to find the probability for
poisson_prob_X1 = poisson.pmf(k1, λ1_15min)

print(f"P(X1 = {k1}) = {poisson_prob_X1}")

P(X1 = 10) = 0.00021572518449585165


Here we can see that the awnser is 3.

#### Question I.2 (2)
Which of the following statements about the expected values of the two random variables is
correct?


In [8]:
E_X1 = λ1_15min # derived from the formula for the Poisson distribution above
E_X2 = 15 * (10 / 60) # rate for 10 cars per hour

ratio = E_X1 / E_X2 # ratio of the expected values
print(f"E[X1] / E[X2] = {ratio}")

E[X1] / E[X2] = 1.0


Here we can see that the awnser is 5.

#### Question I.3 (3)
What is the probability that the time between two cars passing by is greater than 2 minutes
on road 2?

In [11]:
λ2_per_minute = 15 / 60  # rate for road 2 in cars per minute
"""
This code calculates the probability that the time until the next car arrives on road 2 is greater than 2 minutes.
It assumes that the arrival of cars follows a Poisson process with a rate of 15 cars per hour (or 15/60 cars per minute).
Variables:
- λ2_per_minute: The rate of car arrivals on road 2 in cars per minute.
- t: The time period in minutes for which we are calculating the probability.
The probability is calculated using the exponential distribution's survival function:
P(T > t) = exp(-λ * t)
The result is printed in the format: P(T > 2) = <calculated_probability>
"""
t = 2  # time in minutes

prob_greater_than_2_min = exp(-λ2_per_minute * t)
print(f"P(T > 2) = {prob_greater_than_2_min:.4f}")

P(T > 2) = 0.6065


Here we need to round up a little bit, but its still clear that the awser is 3.

## Exercise II
A farm made a study in which 225 chickens were randomly divided into 3 treatment groups
of 75 animals each. Each group were fed with fodder from different feed producers during a
period. For each chicken the weight change over the period of time was measured and the
final data set consists of 225 observations of weight changes. The objective of the study is to
determine if there is statistical evidence for difference in mean weight change for at least one
of the groups. It may be assumed that the variance is the same in all treatment groups.

#### Question II.1 (4)
What kind of statistical analysis is most suitable for this?


*  Multiple linear regression analysis
*  Test for independence in a r × c frequency table (Contingency table)
*  Paired t-tests
*  One-way analysis of variance
*  t-tests

Here the correct awnser is 4.

This is why:
Explanation

Objective
The study aims to compare the **mean weight changes** between three groups (treatment groups with different feed producers).

Number of Groups
Since there are **three groups**, comparing the means using pairwise t-tests (option 5) would not be appropriate. Conducting multiple t-tests increases the risk of a **Type I error** (false positive).

Assumptions for ANOVA
1. **Independence**:  
   The observations (weight changes) for the chickens in each group are independent.

2. **Equal Variance**:  
   It is stated in the problem that the variance is the same in all treatment groups.

3. **Normality**:  
   The weight changes within each group should be approximately normally distributed.  
   *(Note: ANOVA is robust to mild deviations from normality for large sample sizes, like 75 per group.)*

Why ANOVA?
- **ANOVA** is specifically designed to test for differences in the means of three or more groups.
- It determines whether at least one group mean is significantly different from the others.

## Exercise III
The engineers at an international airport have conducted a survey, in which they have timed 40
randomly selected security checks. The average duration of the security checks included in the
survey was 34.66 seconds, and the sample standard deviation was 10.12 seconds, it is assumed
that the times are normally distributed.

#### Question III.1 (5)
Based on the survey, what is the 99% confidence interval for the mean duration of the security
checks?

In [12]:
# Given data
sample_mean = 34.66
sample_std_dev = 10.12
sample_size = 40
confidence_level = 0.99

# Z-score for 99% confidence level
z_score = norm.ppf(1 - (1 - confidence_level) / 2)

# Standard error of the mean
std_error = sample_std_dev / (sample_size ** 0.5)

# Confidence interval
margin_of_error = z_score * std_error
confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)

print(f"99% confidence interval for the mean duration: {confidence_interval}")

99% confidence interval for the mean duration: (30.538383343712, 38.781616656287994)


Here we can see that the awnser is 3.

#### Question III.2 (6)
What is the p-value for the usual test of the null hypothesis H0 : µ = 30 against a two-sided
alternative hypothesis?


In [16]:
# Given data
µ0 = 30 # null hypothesis mean

# Calculate the t-statistic
t_statistic = (sample_mean - µ0) / std_error

# Degrees of freedom
df = sample_size - 1

# Calculate the p-value for a two-sided test
p_value = 2 * (1 - t.cdf(abs(t_statistic), df))

print(f"t-statistic: {t_statistic}")
print(f"p-value: {p_value}")

p_value_percent = p_value * 100 # convert to percentage
print(f"p-value: {p_value_percent:.2f}%")

t-statistic: 2.9122952364396517
p-value: 0.005907860670444576
p-value: 0.59%


With the conversion to percentage we can see that the awnser is 2.  0.59 %

## Exercise IV
A company wants to estimate the cost of producing solar panels. Therefore, the engineers have
designed an experiment to evaluate the costs of producing batches of different sizes and ensure
that the observations are completely independent. The results are given in the table below.

| **Batch Size (units)** | 50   | 100  | 150  | 200  | 250  | 300  | 350  | 400  | 450  | 500  |
|-------------------------|-------|-------|-------|-------|-------|-------|-------|-------|-------|-------|
| **Costs (M DKK)**      | 2.33  | 4.21  | 6.01  | 7.51  | 8.46  | 8.93  | 9.45  | 10.70 | 10.55 | 10.74 |

The data can be read in python using the below code:
// Define the data

Batch = np.arange(1, 11) * 50  # Batch sizes

Costs = np.array([2.33, 4.21, 6.01, 7.51, 8.46, 8.93, 9.45, 10.70, 10.55, 10.74])  # Costs in M DKK


The engineers initially believe that the data can be described by a linear model of the form:

$$
Y_i = \beta_0 + \beta_1 x_i + \varepsilon_i, \quad i \in \{1, \ldots, 10\}
$$

The errors, $\varepsilon_i$, are independent and identically distributed (i.i.d.) with a $\mathcal{N}(0, \sigma^2)$ distribution.

In the model:
- The **response variable** is the **cost** (in M DKK).
- The **explanatory variable** is the **batch size** (in units).

The engineers fit a **linear regression model** using the **least squares method**.

#### Question IV.1 (7)
What proportion of the variation in the costs is explained by the regression model?



In [20]:
Batch = np.arange(1, 11) * 50  # Batch sizes
Costs = np.array([2.33, 4.21, 6.01, 7.51, 8.46, 8.93, 9.45, 10.70, 10.55, 10.74])  # Costs in M DKK

# Fit the linear regression model
model = sm.OLS(Costs, sm.add_constant(Batch)).fit() # Add a constant term to the predictor variable (Batch) to fit the intercept of the linear regression model

# Calculate R-squared
r_squared = model.rsquared
print(f"R-squared: {r_squared}")
r_squared_percent = r_squared * 100 # convert to percentage
print(f"p-value: {r_squared_percent:.2f}%")

R-squared: 0.9059547718379204
p-value: 90.60%


So the awnser here is 2. 90.6%

#### Question IV.2 (8)
What is the 99% confidence interval for the slope, β1?

In [24]:
# Use the data from previous example
alpha = 0.01  # significance level

# Calculate the 99% confidence interval for the slope (β1)
slope_conf_int = model.conf_int(alpha)[1]  # Get the confidence interval for the slope

print(f"99% confidence interval for the slope (β1): [{slope_conf_int[0]:.4f}, {slope_conf_int[1]:.4f}]") 


99% confidence interval for the slope (β1): [0.0112, 0.0251]


So here the awnser is 1.

To validate the model, the engineers make two diagnostic plots: A normal Q-Q plot and a plot

of the residuals against the fitted values. The two plots are seen below:

#### Question IV.3 (9)
Which of the following statements regarding the validity of the model is most correct?

Here the awnser is 4. The residuals vs. fitted values plot indicates that the residuals are not independent of
the fitted values


## Exercise V
The number of persons living in Danish dorms in 2023 is provided by Statistics Denmark. Here
we focus only on a few age-categories:

| Age Group | Males  | Females | Total  |
|-----------|--------|---------|--------|
| 18-24     | 14048  | 14128   | 28176  |
| 25-29     | 8215   | 6028    | 14243  |
| 30-39     | 2735   | 1397    | 4132   |
| **Total** | 24998  | 21553   | 46551  |

#### Question V.1 (10)
What proportion of the residents of Danish dormitories are males (among the 18-39 year olds)?


In [25]:
# Given data
total_males = 14048 + 8215 + 2735
total_residents = 28176 + 14243 + 4132

# Calculate the proportion of males
proportion_males = total_males / total_residents
print(f"Proportion of males: {proportion_males:.4f}")

Proportion of males: 0.5370


Here we can see that the awnser is 4.

#### Question V.2 (11)
We would like to know whether the proportions of males in different age-groups is significantly
different (using a 5% significance level). Which of the following statements is true?


1. We should use a **paired t-test** with $\alpha=0.05$ to test if there is a significant difference between age groups. The result is that we do observe a significant difference.

2. We should use an **unpaired t-test** with $\alpha=0.025$ to test if there is a significant difference between age groups. The result is that we do observe a significant difference.

3. We should use an **unpaired t-test** with $\alpha=0.05$ to test if there is a significant difference between age groups. The result is that we do **NOT** observe a significant difference.

4. We should use a $\chi^2$-test with **6 degrees of freedom** to test if there is a significant difference between age groups. The result is that we do observe a significant difference.

5. We should use a $\chi^2$-test with **2 degrees of freedom** to test if there is a significant difference between age groups. The result is that we do observe a significant difference.

Here the awnser is 5. We should use a $\chi^2$-test with **2 degrees of freedom** to test if there is a significant difference between age groups. The result is that we do observe a significant difference.

In [27]:
# Perform the chi-squared test
chi2_stat, p_val, dof, expected = chi2_contingency(observed)

print(f"Chi-squared statistic: {chi2_stat}")
print(f"p-value: {p_val}")
print(f"Degrees of freedom: {dof}")

# Determine if the result is significant at the 5% significance level
significant = p_val < significance_level
print(f"Significant difference: {significant}")

Chi-squared statistic: 517.1881307484355
p-value: 4.9433840849567903e-113
Degrees of freedom: 2
Significant difference: True


#### Question V.3 (12)
Under the hypothesis that the distribution between for male and females is the same for all
age groups, what is the expected number of males in the age group 18-24 living in dorms (to
be compared with the table above and used for calculating the appropriate test-statistic)?

In [28]:
# Total number of residents in the age group 18-24
total_residents_18_24 = 28176

# Calculate the expected number of males in the age group 18-24
expected_males_18_24 = (total_males / total_residents) * total_residents_18_24
print(f"Expected number of males in the age group 18-24: {expected_males_18_24:.2f}")

Expected number of males in the age group 18-24: 15130.58


Here the awnser is 2.

#### Question V.4 (13)
We now consider only the 18-24 year-olds. Which of the following statements is true (if relevant,
in the answer options, significance level α = 0.05 is used)?

1. Among the 18-24 year-olds, the proportion of males is **NOT** significantly different from 0.5, as the estimated 95% confidence interval for the proportion of males in this age group is \([0.493, 0.504]\).

2. Among the 18-24 year-olds, the proportion of males is **exactly** 0.5.

3. Among the 18-24 year-olds, the proportion of males is significantly different from 0.5, as the estimated 95% confidence interval for the proportion of males in this age group is \([0.501, 0.533]\).

4. Among the 18-24 year-olds, the proportion of males is **NOT** significantly different from 0.5, as the estimated 95% confidence interval for the proportion of males in this age group is \([0.501, 0.533]\).

5. Among the 18-24 year-olds, the proportion of males is significantly different from 0.5, as the estimated 95% confidence interval for the proportion of males in this age group is \([0.532, 0.542]\).



The awnser to this one is 1. Among the 18-24 year-olds, the proportion of males is **NOT** significantly different from 0.5, as the estimated 95% confidence interval for the proportion of males in this age group is \([0.493, 0.504]\).

This can also be solved with the bellow code:

In [32]:
# Given data
males_18_24 = 14048
total_18_24 = 28176
proportion_males_18_24 = males_18_24 / total_18_24

# Standard error for the proportion
std_error_18_24 = (proportion_males_18_24 * (1 - proportion_males_18_24) / total_18_24) ** 0.5

# Z-score for 95% confidence level
z_score_95 = norm.ppf(1 - 0.05 / 2)

# Confidence interval
margin_of_error_18_24 = z_score_95 * std_error_18_24
confidence_interval_18_24 = (proportion_males_18_24 - margin_of_error_18_24, proportion_males_18_24 + margin_of_error_18_24)

print(f"95% confidence interval for the proportion of males in the age group 18-24: ({confidence_interval_18_24[0]:.4f}, {confidence_interval_18_24[1]:.4f})")

95% confidence interval for the proportion of males in the age group 18-24: (0.4927, 0.5044)


#### Question V.5 (14)
Assuming independence between individuals. What is the probability, that 100 or more females
live in a dorm with 190 people, if the probability of an individual being a female is assumed to
be 0.45?

In [34]:
# Given data
n = 190  # total number of people in the dorm
p_female = 0.45  # probability of an individual being a female
k = 100  # number of females

# Calculate the probability using the binomial distribution
prob_100_or_more_females = binom.sf(k - 1, n, p_female)  # sf (survival function) is 1 - cdf

print(f"Probability that 100 or more females live in a dorm with 190 people: {prob_100_or_more_females:.4f}")

Probability that 100 or more females live in a dorm with 190 people: 0.0208


we need to round up a litte but, and here the anwser is 1.

## Exercise VI
A simple predator-prey model is the **Lotka-Volterra model**:

$$
\frac{dx}{dt} = \alpha x - \beta xy
$$

$$
\frac{dy}{dt} = \delta xy - \gamma y
$$

where:
- \( x \) is the size of the prey population.
- \( y \) is the size of the predator population.

The equation allows for a **constant of motion** (i.e., the quantity will stay constant through time for given initial conditions), given by:

$$
K = y^\alpha e^{-\beta y} x^\gamma e^{-\delta x}
$$

Assume that:
- $\alpha = \frac{2}{3} $,
- $ \beta = \frac{4}{3} $,
- $ \gamma = \delta = 1 $.

The predator and prey population sizes have been observed as:
$$
y = \frac{1}{2}, \quad x = 1
$$

The uncertainties of the observations are:
$$
\sigma^2_y = \frac{1}{162}, \quad \sigma^2_x = \frac{1}{8^2}
$$

Furthermore, the observations are assumed to be **independent**.

#### Question VI.1 (15)
Using the non-linear error propagation rule what is the approximation of the variance of K?


In [35]:
import sympy as sp

# Define the variables and constants
x, y = sp.symbols('x y')
alpha = 2/3
beta = 4/3
gamma = 1
delta = 1

# Define the function K
K = y**alpha * sp.exp(-beta * y) * x**gamma * sp.exp(-delta * x)

# Calculate the partial derivatives
partial_K_x = sp.diff(K, x)
partial_K_y = sp.diff(K, y)

# Substitute the given values into the partial derivatives
partial_K_x_val = partial_K_x.subs({x: 1, y: 1/2})
partial_K_y_val = partial_K_y.subs({x: 1, y: 1/2})

# Given variances
sigma_x_squared = 1 / 64
sigma_y_squared = 1 / 162

# Calculate the variance of K
variance_K = (partial_K_x_val**2 * sigma_x_squared) + (partial_K_y_val**2 * sigma_y_squared)
variance_K_val = variance_K.evalf()

print(f"Approximate variance of K: {variance_K_val}")

Approximate variance of K: 0


Here is the awnser 2.

#### Question VI.2 (16)
### Question VI.2 (16)

Assume now that the predator population is observed without error ($\sigma^2_y = 0$), and hence that the only source of uncertainty is the prey population. 

Using the variance from above ($\sigma^2_x = \frac{1}{8^2}$) and assuming normality of the error in $x$, i.e.,

$$
X = x + \varepsilon \quad \text{with} \quad \varepsilon \sim N(0, \sigma^2_x)
$$

In what interval does the standard deviation of $K$ fall? (The answer should not rely on the non-linear approximations of the error propagation rule and should rather be based on simulation.)


In [37]:
# Given data
sigma_x = 1 / 8  # standard deviation of x
num_simulations = 10000  # number of simulations

# Simulate x values with normal distribution
x_samples = np.random.normal(loc=1, scale=sigma_x, size=num_simulations)

# Calculate K for each simulated x value
K_samples = (x_samples * y_val**alpha * np.exp(-x_samples) * np.exp(-beta * y_val))

# Calculate the standard deviation of K
std_dev_K = np.std(K_samples)

print(f"Standard deviation of K: {std_dev_K:.4f}")

# Determine the interval in which the standard deviation falls
if 0.07 <= std_dev_K <= 0.1:
    interval = "(0.07, 0.1)"
elif 0.12 <= std_dev_K <= 0.2:
    interval = "(0.12, 0.2)"
elif 1e-4 <= std_dev_K <= 0.01:
    interval = "(10^-4, 0.01)"
elif 0.03 <= std_dev_K <= 0.05:
    interval = "(0.03, 0.05)"
elif 0.3 <= std_dev_K <= 0.5:
    interval = "(0.3, 0.5)"
else:
    interval = "None of the given intervals"

print(f"The standard deviation of K falls in the interval: {interval}")

Standard deviation of K: 0.0014
The standard deviation of K falls in the interval: (10^-4, 0.01)


The awnser her is: 3

## Exercise VII
Let the function $f(x)$ be defined by:

$$
f(x) = \alpha \phi_1(x) + \beta \phi_2(x),
$$

where $\phi_i(x)$ is the probability density function of a normal random variable with mean $\mu_i$ and variance $\sigma^2_i$.

#### Question VII.1 (17)

Under what conditions is $f(x)$ a probability density function? (The answer should apply for any value of $\sigma_i > 0$ and $\mu_i \in \mathbb{R}$.)

_____SKIPPED______



## Exercise VIII

An aircraft manufacturer uses an expensive type of screws in the production of a certain model.  
To reduce production costs, the manufacturer considers replacing the expensive screws with a cheaper type of screws. Therefore, the manufacturer tests the tensile strength (MPa) of the two types of screws, and the results are shown in the table below:

| **Tensile Strength**       | **Cheap** | **Expensive** |
|----------------------------|-----------|---------------|
| **Sample mean (MPa)**      | 1250      | 1300          |
| **Sample standard deviation (MPa)** | 54.24     | 28.54         |
| **Sample size**            | 25000     | 15000         |

#### Question VIII.1 (18)

Assuming the samples were completely random, what is the 95% confidence interval for the difference in mean tensile strengths (mean of the cheap type minus mean of the expensive type) based on the test results?


In [39]:
# Given data
mean_cheap = 1250
mean_expensive = 1300
std_dev_cheap = 54.24
std_dev_expensive = 28.54
n_cheap = 25000
n_expensive = 15000
confidence_level = 0.95

# Z-score for 95% confidence level
z_score_95 = norm.ppf(1 - (1 - confidence_level) / 2)

# Standard error of the difference in means
std_error_diff = ((std_dev_cheap**2 / n_cheap) + (std_dev_expensive**2 / n_expensive)) ** 0.5

# Confidence interval
mean_diff = mean_cheap - mean_expensive
margin_of_error = z_score_95 * std_error_diff
confidence_interval_diff = (mean_diff - margin_of_error, mean_diff + margin_of_error)

print(f"95% confidence interval for the difference in mean tensile strengths: {confidence_interval_diff}")

95% confidence interval for the difference in mean tensile strengths: (-50.812809165000694, -49.187190834999306)


Here the awnser is 5.

#### Question VIII.2 (19)
Under the null hypothesis H0 : µcheap − µexpensive = −50, what is the observed test statistic
based on the test results?

In [40]:
# Null hypothesis mean difference
null_hypothesis_diff = -50

# Calculate the observed test statistic
observed_test_statistic = (mean_cheap - mean_expensive - null_hypothesis_diff) / std_error_diff

print(f"Observed test statistic: {observed_test_statistic}")

Observed test statistic: 0.0


Here the awnser is 4.

The manufacturer has measured the fuel consumption (logarithmic values, in kg) of the **current model** and the **new model** on 10 popular routes under similar weather and weight conditions.

| Route                         | Current Model | New Model |
|-------------------------------|---------------|-----------|
| Sapporo - Tokyo               | 7.964         | 7.932     |
| Sydney - Melbourne            | 7.813         | 7.762     |
| Mumbai - Delhi                | 8.299         | 8.243     |
| Beijing - Shanghai            | 8.219         | 8.174     |
| Paris - Montreal              | 9.832         | 9.782     |
| Dubai - London                | 9.829         | 9.775     |
| London - New York             | 9.842         | 9.794     |
| New York - Los Angeles        | 9.498         | 9.445     |
| Kuala Lumpur - Singapore      | 7.023         | 6.942     |
| Cancun - Mexico City          | 8.408         | 8.347     |



In [2]:
# Define the data as lists
log_c = [7.964, 7.813, 8.299, 8.219, 9.832, 9.829, 9.842, 9.498, 7.023, 8.408]
log_n = [7.932, 7.762, 8.243, 8.174, 9.782, 9.775, 9.794, 9.445, 6.942, 8.347]

print(f"Logarithm of c: {log_c}")
print(f"Logarithm of n: {log_n}")

Logarithm of c: [7.964, 7.813, 8.299, 8.219, 9.832, 9.829, 9.842, 9.498, 7.023, 8.408]
Logarithm of n: [7.932, 7.762, 8.243, 8.174, 9.782, 9.775, 9.794, 9.445, 6.942, 8.347]


#### Question VIII.3 (20)
What is the p-value for the appropriate test of the null hypothesis H0 : δ = µcurrent−µnew = 0.05
against a two-sided alternative hypothesis? (Here µcurrent and µnew refer to the mean of the
logarithm of the fuel consumption.)


In [7]:


# Calculate the differences
differences = np.array(log_c) - np.array(log_n)

# Perform the paired t-test
t_statistic, p_value = ttest_rel(log_c, log_n)

# Adjust the t-statistic for the null hypothesis difference of 0.05
mean_difference = np.mean(differences)
adjusted_t_statistic = (mean_difference - 0.05) / (np.std(differences, ddof=1) / np.sqrt(len(differences)))

# Calculate the p-value for the adjusted t-statistic
adjusted_p_value = 2 * (1 - t.cdf(abs(adjusted_t_statistic), df=len(differences) - 1))

print(f"Adjusted t-statistic: {adjusted_t_statistic}")
print(f"Adjusted p-value: {adjusted_p_value}")

Adjusted t-statistic: 0.785740937887215
Adjusted p-value: 0.4522084672096742


Here the correct awnser is 3.

## Exercise IX

A car manufacturer wants to determine if there is a difference in breaking strength in beams made with metal from different suppliers. Let Y represent the breaking strength of beams. In
the following breaking strength are measured on beams each made with metal from a single
supplier. Metal from four different suppliers were included in the study and the breaking
strength was measured for 5 similar beams from each supplier:.

| Supplier A | Supplier B | Supplier C | Supplier D |
|------------|------------|------------|------------|
| 92.0       | 131.0      | 74.1       | 90.4       |
| 111.6      | 103.5      | 52.8       | 95.2       |
| 98.4       | 100.0      | 82.5       | 87.6       |
| 87.7       | 84.7       | 94.7       | 63.2       |
| 134.9      | 134.5      | 107.3      | 119.5      |

#### Question IX.1 (21)
The engineers in the company have conducted the following analysis in R. What is the conclusion, at significance level α = 5%, about the difference in breaking strength of the test beams
made with metal from the different suppliers (both conclusion and argument must be correct)?



In [8]:
from scipy.stats import f_oneway

# Breaking strength data for each supplier
supplier_A = [92.0, 111.6, 98.4, 87.7, 134.9]
supplier_B = [131.0, 103.5, 100.0, 84.7, 134.5]
supplier_C = [74.1, 52.8, 82.5, 94.7, 107.3]
supplier_D = [90.4, 95.2, 87.6, 63.2, 119.5]

# Perform one-way ANOVA
f_statistic, p_value = f_oneway(supplier_A, supplier_B, supplier_C, supplier_D)

print(f"F-statistic: {f_statistic}")
print(f"p-value: {p_value}")

# Conclusion at significance level α = 0.05
alpha = 0.05
if p_value < alpha:
    conclusion = "There is a significant difference in breaking strength between the suppliers."
else:
    conclusion = "There is no significant difference in breaking strength between the suppliers."

print(conclusion)

F-statistic: 2.026965545909903
p-value: 0.150669891510969
There is no significant difference in breaking strength between the suppliers.
